ui/src/app/machines/search.js

```diff
@@ -20,7 +20,7 @@ export const getCurrentFilters = (search) => {
   // Match filters with parens e.g. 'status:(new,deployed)'.
   // Then: match filters without parens e.g. 'status:new,deployed' or 'status'
   const filterMatchingRegex = search.matchAll(
-    /(\b\w+:!*\([^)]+\))|(!*\w+\S*)/g
+    /(\b[\w-]+:!*\([^)]+\))|(!*\w+\S*)/g
```

Comment:
Amended regex to include `-`s, which is required for the workload filters that take the form `workload-${type}`.

Reply:
Nice. I reckon this needs a test as well.